skrf/vectorFitting.py

```diff
@@ -0,0 +1,765 @@
+import numpy as np
+import os
+import matplotlib.pyplot as mplt
+from matplotlib.ticker import EngFormatter
+
+
+class VectorFitting:
+    """
+    This is a Python implementation of the Vector Fitting algorithm [1]. The code is based on two improvements for
+    relaxed pole relocation [2] and efficient (fast) solving [3]. See also the Vector Fitting website [4] for further
+    information and download of the papers listed below. A Matlab implementation is also available there for reference.
+
+    [1] B. Gustavsen, A. Semlyen, "Rational Approximation of Frequency Domain Responses by Vector Fitting", IEEE
+        Transactions on Power Delivery, vol. 14, no. 3, pp. 1052-1061, July 1999
+    [2] B. Gustavsen, "Improving the Pole Relocating Properties of Vector Fitting", IEEE Transactions on Power Delivery,
+        vol. 21, no. 3, pp. 1587-1592, July 2006
+    [3] D. Deschrijver, M. Mrozowski, T. Dhaene, D. De Zutter, "Marcomodeling of Multiport Systems Using a Fast
+        Implementation of the Vector Fitting Method", IEEE Microwave and Wireless Components Letters, vol. 18, no. 6,
+        pp. 383-385, June 2008
+    [4] https://www.sintef.no/projectweb/vectorfitting/
+    """
+
+    def __init__(self, network):
```

Comment:
Would it be useful to be able to construct the `VectorFitting` object from given poles and zeros? (just a question, I have no idea of potential use cases)

Reply:
I don't think that's a good idea. The model parameters are strongly related to a specific set of frequency responses, i.e. a `Network`, and most class functions use both the model parameters and the `Network` attributes, such as the frequency vector `Network.f` and the number of ports `Network.n_ports`. In fact, it is already pretty risky to allow the import of model parameters via `read_npz()`. Maybe it is even better to remove that function altogether and only allow the export via `write_npz()`?